Group6PublicLibrary/src/group6ranger/RangeSensor.java

```diff
@@ -0,0 +1,145 @@
+package group6ranger;
+
+import java.util.*;
+import org.iot.raspberry.grovepi.GrovePi;
+import org.iot.raspberry.grovepi.devices.GroveUltrasonicRanger;
+
+/**
+ * Abstract subject provides methods for storing, removing, and notifying observers
+ */
+public abstract class RangeSensor{
+    private List<IRangeObserver> observers = new ArrayList<IRangeObserver>();
+    GroveUltrasonicRanger ranger;
```

Comment:
Should be private or protected, if elements need access to it beyond these contexts they should be doing so via accessors, but only if necessary